data/multiplayer/scenarios/4p_King_of_the_Hill.cfg

```diff
@@ -4,20 +4,44 @@
     [store_side]
         side={SIDE}
     [/store_side]
-    {RANDOM $side.recruit}
-    [store_unit_type]
-        type=$random
-    [/store_unit_type]
-    # If the unit is level 0 then switch to one of its advancements instead.
-    [if]
-        [variable]
-            name=unit_type.level
-            equals=0
-        [/variable]
-        [then]
-            {RANDOM $unit_type.advances_to}
-        [/then]
-    [/if]
+    # Get the faction leader list of the side.
+    [lua]
+        code=<<
+            local t = ...
+            local era = wesnoth.get_era(wesnoth.scenario.era.id)
+            for i,v in ipairs(era) do
+                if v["tag"] == "multiplayer_side" then
+                    for k,faction in pairs(v) do
+                        if faction["id"] == t.faction then
```

Comment:
Even if you wanted to use `ipairs` for some reason, this isn't correct. You would write `local faction = v.contents` (or `v["contents"]` if you prefer) and go from there.

Reply:
Was attempting to loop over only the integer indexed subtags and skip over the string 'indexed' attributes. But your improvement of using `wml.child_range` makes this little attempted optimization obsolete.